### PR TITLE
Implement hooks in XDG_CONFIG_HOME by default

### PR DESCRIPTION
**Pull request by jdorel**
#### Description

Implement hooks in XDG_CONFIG_HOME by defalt

#### Additional information...

- [X] Have you run the test suite?

Currently one problem : 
```
Failed:                        
version.t                           1
```
No idea of how to resolve it

### CODE REVIEW

**Comment by tbabej** [comment]
Hi @jdorel, thanks for looking into this!

The `version.t` problem just means the tests were build with older "commit" than the task binary being tested itself. It serves more as a warning - as you can see, in the CI all the platforms successfully passed.

**Review by tbabej — CHANGES_REQUESTED** [review]

File: src/util.h [hunk 1]
```diff
@@ -65,4 +65,9 @@ const char* optionalBlankLine ();
 void setHeaderUnderline (Table&);
 
 #endif
+
+const std::string getXdgConfigHome ();
+const std::string getDefaultHooksLocation();
```
[the remark this answers is not in the document]
Comment (jdorel) [reply src/util.h:70]: Indeed

File: src/util.cpp [hunk 2]
```diff
@@ -309,3 +310,28 @@ void setHeaderUnderline (Table& table)
...
+  if (xdg_config_home.back () == '/')
+    xdg_config_home.pop_back();
+
+  return xdg_config_home;
+}
+
+////////////////////////////////////////////////////////////////////////////////
+const std::string getDefaultHooksLocation()
+{
+  return format ("{1}/task/hooks", getXdgConfigHome());
```
[the remark this answers is not in the document]
Comment (jdorel) [reply src/util.cpp:334]: I implemented the logic in `Hooks.cpp`, but you are right that it should be implemented in `utils.cpp`,

[the remark this answers is not in the document]
Comment (jdorel) [reply src/util.cpp:334]: Maybe the full implementation should be in a function `getHooksLocation` in `Hooks.cpp` or in `Context.cpp` ?

File: src/Hooks.cpp [hunk 3]
```diff
@@ -72,8 +72,15 @@ void Hooks::initialize ()
   }
   else
   {
-    d = Directory (Context::getContext ().config.get ("data.location"));
-    d += "hooks";
+    if (Context::getContext ().rc_file.exists ())
```
[the remark this answers is not in the document]
Comment (jdorel) [reply src/Hooks.cpp:75]: I didn't want to hardcode the path in two different location, but I didn't realize that `rc_file` was modified during context initialization.

**Comment by tbabej** [comment]
Looks like the latest set of changes here broke the tests for hooks :crying_cat_face:

File: src/commands/CmdShow.cpp [hunk 4]
```diff
@@ -240,7 +240,7 @@ int CmdShow::execute (std::string& output)
           i.first.substr (0,  8) != "holiday."              &&
           i.first.substr (0,  7) != "report."               &&
           i.first.substr (0,  6) != "alias."                &&
-          i.first.substr (0,  5) != "hook."                 &&
```
Comment (jdorel) [inline src/commands/CmdShow.cpp:243]: @tbabej Fixed an error in config verification

File: test/basetest/task.py [hunk 5]
```diff
@@ -52,8 +53,9 @@ def __init__(self, taskd=None, taskw=DEFAULT_TASK):
 
         with open(self.taskrc, 'w') as rc:
             rc.write("data.location={0}\n"
+                     "hooks.location={1}\n"
```
Comment (jdorel) [inline test/basetest/task.py:56]: @tbabej Fix test suite : specify `hooks.location`, otherwise search in ~/.config/task

**Review by jdorel — COMMENTED** [review]
@tbabej Updated the test suite + fixed a bug

**Comment by djmitche** [comment]
@jdorel would you like to rebase this and we can get it merged?

**Pull request closed without merging**